src/embed_tests/TestPyObject.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using Python.Runtime;
+
+namespace Python.EmbeddingTest
+{
+    public class TestPyObject
+    {
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            PythonEngine.Initialize();
+        }
+
+        [OneTimeTearDown]
+        public void Dispose()
+        {
+            PythonEngine.Shutdown();
+        }
+
+        [Test]
+        public void TestGetDynamicMemberNames()
+        {
+            List<string> expectedMemberNames = new List<string>
+            {
+                "__class__",
+                "__delattr__",
+                "__dict__",
+#if PYTHON3
+                "__dir__",
+#endif
+                "__doc__",
+#if PYTHON3
+                "__eq__",
+#endif
+                "__format__",
+#if PYTHON3
+                "__ge__",
+#endif
+                "__getattribute__",
+#if PYTHON3
+                "__gt__",
+#endif
+                "__hash__",
+                "__init__",
+#if PYTHON36
+                "__init_subclass__",
+#endif
+#if PYTHON3
+                "__le__",
+                "__lt__",
+#endif
+                "__module__",
+#if PYTHON3
+                "__ne__",
+#endif
+                "__new__",
```

Comment:
The defines here make the whole thing quite brittle, I think, to me it would be perfectly fine to just check that the explicitly added members exist (i.e. `add`, `getNumber` etc.).